File: Java/FizzBuzz2.java
public class FizzBuzz2 {
    static final int LIMIT = 100;

    public static void main(String[] args) {
        for (int i = 1; i <= LIMIT; i++) {
            switch ((i % 3 == 0 ? 1 : 0) | (i % 5 == 0 ? 1 : 0) << 1) {
                case 0:
                    System.out.print(i);
                    break;
                case 1:
                    System.out.print("Fizz");
                    break;
                case 2:
                    System.out.print("Buzz");
                    break;
                case 3:
                    System.out.print("FizzBuzz");
                    break;
            }
            if (i < LIMIT) {
                System.out.print(", ");
            }
        }
        System.out.println();
    }
}

File: Erlang/fizzbuzz2.erl
-module(fizzbuzz2).
-export([main/0, main/1]).

-define(LIMIT, 100).

get_str(0, L) -> L;
get_str(X, L) when X rem 15 == 0 -> get_str(X - 1, ["FizzBuzz" | L]);
get_str(X, L) when X rem 3 == 0 -> get_str(X - 1, ["Fizz" | L]);
get_str(X, L) when X rem 5 == 0 -> get_str(X - 1, ["Buzz" | L]);
get_str(X, L) -> get_str(X - 1, [integer_to_list(X) | L]).


main() ->
    io:fwrite("~s~n", [lists:flatten(lists:join(", ", get_str(?LIMIT, [])))]).

main(_) -> main().
